classify-split-extract-workflow/classify-job/split_and_classify.py

```diff
@@ -0,0 +1,358 @@
+        for doc_uri in input_uris
+    ]
+    gcs_documents = documentai.GcsDocuments(documents=input_docs)
+    input_config = documentai.BatchDocumentsInputConfig(gcs_documents=gcs_documents)
+
+    gcs_output_uri = f"gs://{DOCAI_OUTPUT_BUCKET}"
+    timestamp = utils.get_utc_timestamp()
+    gcs_output_uri_prefix = "classifier_out_" + timestamp
+    destination_uri = f"{gcs_output_uri}/{gcs_output_uri_prefix}/"
+    output_config = documentai.DocumentOutputConfig(
+        gcs_output_config={"gcs_uri": destination_uri}
+    )
+
+    logger.info(f"input_config = {input_config}, output_config = {output_config}")
+    logger.info(
+        f"Calling DocAI API for {len(input_uris)} document(s) "
+        f"using {processor.display_name} processor "
+        f"type={processor.type_}, path={processor.name}"
+    )
+
+    request = documentai.types.document_processor_service.BatchProcessRequest(
+        name=processor.name,
+        input_documents=input_config,
+        document_output_config=output_config,
+    )
+    operation = dai_client.batch_process_documents(request)
+
+    logger.info(f"Waiting for operation {operation.operation.name} to complete...")
+    operation.result()
+
+    metadata: BatchProcessMetadata = documentai.BatchProcessMetadata(operation.metadata)
+    return process_classify_results(metadata)
+
+
+def process_classify_results(metadata: BatchProcessMetadata) -> Optional[Dict]:
+    """Processes the results of a classification operation."""
+    logger.info(f"handling classification results - operation.metadata={metadata}")
+
+    documents = {}
+
+    if metadata.state != documentai.BatchProcessMetadata.State.SUCCEEDED:
+        raise ValueError(f"Batch Process Failed: {metadata.state_message}")
+
+    for process in metadata.individual_process_statuses:
+        matches = re.match(r"gs://(.*?)/(.*)", process.output_gcs_destination)
+        output_bucket, output_prefix = matches.groups()
+        input_gcs_source = process.input_gcs_source
+
+        logger.info(
+            f"output_bucket = {output_bucket}, "
+            f"output_prefix={output_prefix}, "
+            f"input_gcs_source = {input_gcs_source}, "
+            f"output_gcs_destination = {process.output_gcs_destination}"
+        )
+
+        # TODO add support for shards using toolbox, after issue below is addressed:
+        #  https://github.com/googleapis/python-documentai-toolbox/issues/332
+        output_blob = list(storage_client.list_blobs(output_bucket, prefix=output_prefix + "/"))[0]
+
+        if ".json" not in output_blob.name:
+            logger.info(
+                f"Skipping non-supported file: {output_blob.name} - Mimetype: {output_blob.content_type}"
+            )
+            continue
+
+        document_out = documentai.Document.from_json(
+            output_blob.download_as_bytes(), ignore_unknown_fields=True
+        )
+        blob_entities = document_out.entities
```

Comment:
Most of this can be done with Document AI Toolbox

Reply:
When new revision will be available for pip install, I will use it. But before the fix, i cannot use it.

Reply:
Right now the latest available version is still google-cloud-documentai-toolbox==0.13.5a0

Reply:
I understand. There's currently a release freeze blocking a new version from releasing.